peerconnection.go

```diff
@@ -1887,3 +2191,57 @@ func (pc *PeerConnection) generateMatchedSDP(useIdentity bool, includeUnmatched
 
 	return populateSDP(d, detectedPlanB, pc.api.settingEngine.candidates.ICELite, pc.api.mediaEngine, connectionRole, candidates, iceParams, mediaSections, pc.ICEGatheringState())
 }
+
+func (pc *PeerConnection) updateExtMaps(weOffer bool) error {
+	remoteExtMaps := map[int]*sdp.ExtMap{}
+
+	// populate the extmaps from the current remote description
+	for _, media := range pc.RemoteDescription().parsed.MediaDescriptions {
+		// populate known remote extmap and handle conflicts.
+		for _, attr := range media.Attributes {
+			if attr.Key != "extmap" {
+				continue
+			}
+			em := &sdp.ExtMap{}
+			if err := em.Unmarshal("extmap:" + attr.Value); err != nil {
+				pc.log.Warnf("Failed to parse ExtMap: %v", err)
+				continue
+			}
+			if remoteExtMap, ok := remoteExtMaps[em.Value]; ok {
+				if remoteExtMap.Value != em.Value {
+					return fmt.Errorf("RemoteDescription changed some extmaps values")
```

Comment:
I haven't dug into the spec, how common is it that possibly different media descriptions can have the same extmap URI with a different ID?

Reply:
Chrome and Firefox uses totally different values.

Reply:
What I'm saying is, if I send two video tracks in the same offer (maybe with different codecs), can you guarantee that the `extmap` for `urn:ietf:params:rtp-hdrext:sdes:rtp-stream-id` has the same ID on both of them (i.e. a common value across the entire offer)? Or might the `extmap` IDs for the same extension on different media descriptions be different even in the same offer?

Reply:
@cretz The spec https://tools.ietf.org/html/rfc5285 says:
```
The mapping may be provided per media stream (in the media-level
   section(s) of SDP, i.e., after an "m=" line) or globally for all
   streams (i.e., before the first "m=" line, at session level).  The
   definitions MUST be either all session level or all media level; it
   is not permitted to mix the two styles.  In addition, as noted above,
   the IDs used MUST be unique for each stream type for a given media,
   or for the session for session-level declarations.
```
So the last phrase: **In addition, as noted above, the IDs used MUST be unique for each stream type for a given media, or for the session for session-level declarations.** make it intends that ID (extmap values) may be different for different media streams.

The conflict is that the sdes mid extension header value cannot be different since it's used for determining the media section...
Currently chrome and firefox put extmaps per media section (not globally) and use the same IDs for all the media sections.

Let me know if you prefer that I split extmaps for media sections with an additional check to ensure that the sdes-mid extension value is the same for all the media section.

Reply:
@cretz I'll change this patch to store and negotiate them per transceiver level since it's cleaner and useful for future improvements.